ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+		}
+	}
+
+	return keyValuePairs, secrets, createSecrets, nil
+}
+
+// exportSecrets extracts all secrets and environment variables from ECS creation options
+func (m *BasicECSPodCreator) exportSecrets(ctx context.Context, merged cocoa.ECSPodCreationOptions) ([]*ecs.KeyValuePair, []cocoa.PodSecret, []cocoa.NamedSecret, error) {
+	allEnvVars := []*ecs.KeyValuePair{}
+	allSecrets := []cocoa.PodSecret{}
+	allSecretsToCreate := []cocoa.NamedSecret{}
+
+	for _, def := range merged.ContainerDefinitions {
+		envVars, secrets, secretsToCreate, err := m.exportEnvVars(ctx, def.EnvVars)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		allEnvVars = append(allEnvVars, envVars...)
+		allSecrets = append(allSecrets, secrets...)
+		allSecretsToCreate = append(allSecretsToCreate, secretsToCreate...)
+	}
+
+	return allEnvVars, allSecrets, allSecretsToCreate, nil
+}
+
+// createSecrets creates secrets that do not already exist
+func (m *BasicECSPodCreator) createSecrets(ctx context.Context, secrets []cocoa.NamedSecret) error {
+	for _, secret := range secrets {
+		_, err := m.vault.CreateSecret(ctx, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// translateSecrets translates a PodSecret to an ECS Secret
+func translateSecrets(ctx context.Context, secrets []cocoa.PodSecret) []*ecs.Secret {
+	ecsSecrets := []*ecs.Secret{}
+
+	for _, secret := range secrets {
+		ecsSecret := ecs.Secret{}
+		ecsSecret.SetName(*secret.Name)
+		ecsSecret.SetValueFrom(*secret.Value)
+		ecsSecrets = append(ecsSecrets, &ecsSecret)
+	}
+
+	return ecsSecrets
+}
+
+// exportPodCreationOptions converts options to create a pod into its equivalent ECS task definition.
+func (m *BasicECSPodCreator) exportPodCreationOptions(ctx context.Context, merged cocoa.ECSPodCreationOptions, envVars []*ecs.KeyValuePair, secrets []cocoa.PodSecret) (*ecs.RegisterTaskDefinitionInput, error) {
+	var containerDefs []*ecs.ContainerDefinition
+
+	for _, def := range merged.ContainerDefinitions {
+		if def.CPU == nil {
+			return nil, errors.New("missing CPU in container definition")
+		}
+		cpu64 := int64(*def.CPU)
+
+		if def.MemoryMB == nil {
+			return nil, errors.New("missing MemoryMB in container definition")
+		}
+		mem64 := int64(*def.MemoryMB)
+
+		containerDef := ecs.ContainerDefinition{}
+		containerDef.SetCommand(utility.ToStringPtrSlice(def.Command)).
+			SetCpu(cpu64).
+			SetImage(*def.Image).
+			SetName(*def.Name).
+			SetMemory(mem64).
+			SetEnvironment(envVars).
+			SetSecrets(translateSecrets(ctx, secrets))
+
+		containerDefs = append(containerDefs, &containerDef)
+	}
+
+	if merged.MemoryMB == nil {
+		return nil, errors.New("missing MemoryMB definition")
+	}
+
+	if merged.CPU == nil {
+		return nil, errors.New("missing CPU definition")
+	}
+
+	taskDef := &ecs.RegisterTaskDefinitionInput{}
+	taskDef.SetContainerDefinitions(containerDefs).
+		SetMemory(strconv.Itoa(*merged.MemoryMB)).
+		SetCpu(strconv.Itoa(*merged.CPU)).
+		SetTaskRoleArn(*merged.TaskRole).
```

Comment:
This is not guaranteed to be set. Use `utility.FromStringPtr` and only set it if it's non-empty.
```go
taskDef.SetTaskRoleArn(utility.FromStringPtr(merged.TaskRole)
```